src/__tests__/tigris.rpc.spec.ts

```diff
@@ -344,58 +347,34 @@ describe("rpc tests", () => {
 		return readOnePromise;
 	});
 
-	it("findManyStream", (done) => {
+	it("findManyStream", async () => {
 		const tigris = new Tigris({serverUrl: "0.0.0.0:" + SERVER_PORT, insecureChannel: true});
 		const db1 = tigris.getDatabase("db3");
 		let bookCounter = 0;
-		let success = true;
-		success = true;
-		db1.getCollection<IBook>("books").findManyStream({
+		const booksIterator = db1.getCollection<IBook>("books").findManyStream({
 			op: SelectorFilterOperator.EQ,
 			fields: {
 				author: "Marcel Proust"
 			}
-		}, {
-			onEnd() {
-				// test service is coded to return 4 books back
-				expect(bookCounter).toBe(4);
-				expect(success).toBe(true);
-				done();
-			},
-			onNext(book: IBook) {
-				bookCounter++;
-				expect(book.author).toBe("Marcel Proust");
-			},
-			// eslint-disable-next-line @typescript-eslint/no-unused-vars
-			onError(_error: Error) {
-				success = false;
-			}
 		});
+
+		for await (const book of booksIterator) {
```

Comment:
looks much nicer. 

In this model we either return empty iterator if there is no data or if there is an error?

Reply:
User will have to wrap the for loop inside a try catch

Reply:
This is same as:
```
try{
await somePromise
} catch {
}
```
or promise.then().catch()

Reply:
great!